File: typescript/resistor-color/resistor-color.ts
export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
] as const;

type Colors = typeof COLORS[number];

export function colorCode(color : Colors) : number {
  return COLORS.indexOf(color);
}

File: kotlin/resistor-color/src/main/kotlin/ResistorColor.kt
object ResistorColor {
  private val colorMap = mapOf(
    "black" to 0,
    "brown" to 1,
    "red" to 2,
    "orange" to 3,
    "yellow" to 4,
    "green" to 5,
    "blue" to 6,
    "violet" to 7,
    "grey" to 8,
    "white" to 9
  )

  fun colorCode(input: String) = colorMap.getValue(input)
  fun colors() = colorMap.keys.toList()
}
